### DISCUSSION TITLE
Intercept not working properly

### DISCUSSION THREAD

**Discussion by B34v0n** [cypress-io/cypress#27251] [Questions and Help] [answered]
I'm trying to test my Angular app in the component-testing-mode.
In there, I've got a component with a service with 2 requests.
One of them can be intercepted, the other one can't, and I cannot figure out, why.

So here are the details:
In the service, there are the following two HTTP-Request:
`this.http.post<Type>(url + '/catalog', body, { headers });` 
and
`this.http.get<Type>(url + '/measure/' + id);`
(The Type is the same)

The Interceptors look like this:
`cy.intercept({ method: 'POST', url: '**/catalog' }, [returnValue]).as('getCatalog');`
and
`cy.intercept({ method: 'GET', url: '**/measure/**' }, [returnValue]).as('getMeasure');`

The Post-Interceptor works like a charm and always returns the returnValue.
BUT the Get-Interceptor doesn't work at all. Cypress says, the interceptor was never called, but at the end of the test also says, that there is a HTTP-Request, that returned 404. But said request is `.../url/measure/1`. I even tried to copy the complete url, but that also didn't work.

Anybody got an idea, what the problem could be? I'm desperate.

**Comment by B34v0n** [comment, marked as answer]
If anyone has the same error and is desperate as I was, I had a stupid error.
My URL started without a slash.
The requests were send to the cypress-own-backend.
So the post-Request was mocked, because there was no post in the cypress-backend, but the get-Request wasn't mocked, because cypress tried to get something from this backend.

so instead of `const url = 'this/is/my/url'` I just needed to write `const url = '/this/is/my/url'`

**Comment by DeniDoman** [comment]
Thank you for the hint. But there is still a problem when an app actually fetches url without slash (e.g. `fetch(".auth/me")`) - Cypress can't intercept it, returning the internal source code. 

Here the issue: https://github.com/cypress-io/cypress/issues/31215
